### PR TITLE
feat(web): list view for version images

### PR DESCRIPTION
**Pull request by robot9706**
Added a list view for version images.
Added a toggle to select between tile and list view.
Added a modal to select the image tag in list mode.

Added createdBy, createdAt, updatedBy, updatedAt to images.

Version page defaults to "list" view instead of "tile" view.
Image editor card defaults to "config" page instead of "tag" page.
Changed image tag fetching to on demand (this should reduce the demand on the DockerHub API).

### CODE REVIEW

File: web/crux-ui/src/components/products/versions/version-images-section.tsx [hunk 1]
```diff
@@ -1,46 +1,165 @@
...
   return images.length ? (
-    <DyoWrap itemClassName="xl:w-1/2 py-2">
-      {images
-        .sort((one, other) => one.order - other.order)
-        .map((it, index) => {
-          const key = imageTagKey(it.registryId, it.name)
-          const details = imageTags[key]
-
-          return (
+    viewMode === 'tile' ? (
```
Comment (m8vago) [inline web/crux-ui/src/components/products/versions/version-images-section.tsx:104]: please make components for the two viewMode

File: web/crux/src/app/image/image.mapper.ts [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 import { ContainerConfig as ProtoContainerConfig, ImageResponse } from 'src/grpc/protobuf/proto/crux'
 import { UniqueKeyValue } from 'src/shared/model'
 
 @Injectable()
 export default class ImageMapper {
-  toGrpc(image: ImageWithConfig): ImageResponse {
+  toGrpc(image: ImageWithConfig, registryName?: string): ImageResponse {
     return {
       ...image,
+      registryName: registryName || '',
```
Comment (m8vago) [inline web/crux/src/app/image/image.mapper.ts:13]: smart

File: web/crux/src/app/image/image.service.ts [hunk 3]
```diff
@@ -104,8 +105,26 @@ export default class ImageService {
 
     this.imagesAddedToVersionEvent.next(images)
 
+    const registries = await this.prisma.registry.findMany({
```
Comment (m8vago) [inline web/crux/src/app/image/image.service.ts:108]: i think you can add the registry into the images.create() call's include section, so you don't need to query the regs

**Review by polaroi8d — APPROVED** [review]

File: web/crux/src/app/image/image.service.ts [hunk 4]
```diff
@@ -104,8 +106,15 @@ export default class ImageService {
 
     this.imagesAddedToVersionEvent.next(images)
 
+    const registryLookup = images.reduce((prev, current) => {
```
Comment (m8vago) [inline web/crux/src/app/image/image.service.ts:109]: we dont need a lookup. the registry is there in the image

Comment (robot9706) [reply web/crux/src/app/image/image.service.ts:109]: oh 🙈

**Review by m8vago — APPROVED** [review]

**Pull request merged**